src/lib.rs

```diff
@@ -16,36 +16,41 @@ pub mod config;
 
 use crate::cli::CliArgs;
 use crate::error::Result;
+use crate::helper::args::FAIL_EMOTICON;
 use cli::CliCommands;
 use config::Config;
+use console::style;
 use helper::args::get_args_help;
 use helper::docs::get_docs_help;
 use std::io::Write;
 
 /// Runs `halp`.
-pub fn run<Output: Write>(mut cli_args: CliArgs, output: &mut Output) -> Result<()> {
-    let config = if let Some(config_file) = cli_args
+pub fn run<Output: Write>(cli_args: CliArgs, output: &mut Output) -> Result<()> {
+    let mut config = if let Some(config_file) = cli_args
         .config
         .to_owned()
         .or_else(Config::get_default_location)
     {
-        let config = Config::parse(&config_file)?;
-        config.update_args(&mut cli_args);
-        Some(config)
+        Config::parse(&config_file)?
     } else {
-        None
+        let config = Config::default();
+
+        if let Err(e) = config.write() {
+            eprintln!(
+                "{} Failed to write default config: {}",
+                style(FAIL_EMOTICON).for_stderr().yellow(),
+                e
+            );
+        }
+        config
```

Comment:
I just thought about this and I'm not sure if it is okay to write the config without even notifying the user. What about we add an command line option for this such as `--write-config`?

Reply:
> I just thought about this and I'm not sure if it is okay to write the config without even notifying the user.

I think isn't a big deal most of the programs doing this for a decade

Reply:
@orhun I hope you'll respond soon because I really wanna get this PR merged

Reply:
> I think this isn't a big deal, most of the programs doing this for a decade

Fair enough.

> I hope you'll respond soon because I really wanna get this PR merged

Sorry I was on a mini vacation and this was in my backlog.